src/utils.ts

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-template-curly-in-string, @typescript-eslint/no-explicit-any */
+ 
+import { homedir } from 'node:os'
+import path from 'node:path'
+import process from 'node:process'
+
+import vscode, { Uri } from 'vscode'
+
+/**
+ * code migrated from: https://github.com/tjx666/open-in-external-app/blob/master/src/parseVariables.ts
+ * 
+ * @see https://code.visualstudio.com/docs/editor/variables-reference
+ */
+export async function parseVariables(str: string, activeFile: Uri) {
+    const replacement: Map<string | RegExp, string | ((substring: string, ...args: any[]) => string) | undefined> =
+        new Map([
+            ['${userHome}', homedir()],
+            ['${pathSeparator}', path.sep],
+        ])
+
+    const { workspaceFolders } = vscode.workspace
+    if (workspaceFolders?.length) {
+        const workspaceFolder = workspaceFolders[0]!
+        replacement.set('${workspaceFolder}', workspaceFolder.uri.fsPath)
+        replacement.set('${workspaceFolderBasename}', workspaceFolder.name)
+    }
+
+    const { activeTextEditor } = vscode.window
+    const absoluteFilePath = activeFile.fsPath
+    replacement.set('${file}', absoluteFilePath)
+
+    const activeWorkspace = activeFile ? vscode.workspace.getWorkspaceFolder(activeFile) : undefined
+    replacement.set('${fileWorkspaceFolder}', activeWorkspace?.uri.fsPath)
+    replacement.set('${cwd}', activeWorkspace?.uri.fsPath)
+
+    let relativeFilePath: string | undefined
+    if (absoluteFilePath && activeWorkspace) {
+        relativeFilePath = absoluteFilePath.replace(activeWorkspace.uri.fsPath, '').slice(path.sep.length)
+        replacement.set('${relativeFile}', relativeFilePath)
```

Comment:
I think this one I'll be using a lot, don't like having long paths entered in terminal